src/features/core/deviceModeTransformation/util.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * A helper function that will take rudderEelement as an event and generate
+ * a batch payload that will be sent to transformation server
+ *
+ */
+const createPayload = (event, token) => {
+  const orderNo = Date.now();
+  const payload = {
+    metadata: {
+      "Custom-Authorization": token
+    },
+    batch: [
```

Comment:
is it intented that the batch always have 1 item? is thsi how mobiles also do it? do we have limitation from server side?

Reply:
Yes, intentionally we did it.
For mobiles it is different. Mobile SDKs have a database with two tables to manage this.
No limitation from the server.

Reply:
got it